### PR TITLE
[zh-cn] update 'The box model'

### LINKED ISSUE #17757: [zh-cn] update `the box model` translations
https://github.com/mdn/translated-content/blob/main/files/zh-cn/learn/css/building_blocks/the_box_model/index.md

It's a bit outdated.

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

Update 'The box model'

### Related issues and pull requests

fixes #17757 
part of #12632

### CODE REVIEW

File: files/zh-cn/learn/css/building_blocks/the_box_model/index.md [hunk 1]
```diff
@@ -253,62 +272,64 @@ html {
...
+**在带有 `.links-list a` 选择器的规则中添加 `display: inline-block`，你就会看到它是如何通过使其他元素考虑内边距来解决这个问题的。**
 
-这种情况在导航栏中很常见。下面的导航使用 flexbox 显示在一行中，我们为 `<a>` 元素添加了内边距，因为我们希望能够在 `<a>` 在鼠标移动到上面时改变背景色。内边距似乎覆盖了 `<ul>` 元素上的边框。这是因为 `<a>` 是一个内联元素。
+{{EmbedGHLiveSample("css-examples/learn/box-model/inline-block-nav.html", '100%', 700)}}
 
-使用 `.links-list a` 选择器将 `display: inline-block` 添加到样式规则中，你将看到它是如何通过内边距推开其他元素来修复这个问题的。
+## 技能测试！
 
-{{EmbedGHLiveSample("css-examples/learn/box-model/inline-block-nav.html", '100%', 600)}}
+本文已经结束，你还记得最重要的信息吗？在继续学习之前，可以找到一些进一步的测试来验证您是否保留了这些信息——请参阅[技能测试：盒模型](/zh-CN/docs/Learn/CSS/Building_blocks/Box_Model_Tasks)。
```
[the remark this answers is not in the document]
Comment (jasonren0403) [reply files/zh-cn/learn/css/building_blocks/the_box_model/index.md:327]:
```suggestion
本文已经结束，你还记得最重要的信息吗？在继续学习之前，可以找到一些进一步的测试来验证你是否保留了这些信息——请参阅[技能测试：盒模型](/zh-CN/docs/Learn/CSS/Building_blocks/Box_Model_Tasks)。
```

File: files/zh-cn/learn/css/building_blocks/the_box_model/index.md [hunk 2]
```diff
@@ -1,244 +1,263 @@
...
+{{EmbedGHLiveSample("css-examples/learn/box-model/box-models.html", '100%', 1100)}}
 
-> **备注：** [单击此处查看答案](https://github.com/mdn/css-examples/blob/master/learn/solutions.md#the-box-model)。
+> **备注：** 在[这里](https://github.com/mdn/css-examples/blob/master/learn/solutions.md#the-box-model)查看该任务的解答。
 
-### 使用调试工具来查看盒模型
+### 使用浏览器调试工具来查看盒模型
 
-[浏览器开发者工具](/zh-CN/docs/Learn/Common_questions/What_are_browser_developer_tools)可以使你更容易地理解 box 模型。如果你在 Firefox 的 DevTools 中查看一个元素，你可以看到元素的大小以及它的外边距、内边距和边框。这是一个很好的检查元素大小的方式，可以便捷的判断你的盒子大小是否符合预期 !
+[浏览器开发者工具](/zh-CN/docs/Learn/Common_questions/Tools_and_sestup/What_are_browser_developer_tools)可以使你更容易地理解盒模型。如果你在 Firefox 的 DevTools 中查看一个元素，你可以看到元素的大小以及它的外边距、内边距和边框。这是一个很好的检查元素大小的方式，可以便捷的判断你的盒子大小是否符合预期！
```
Comment (jasonren0403) [inline files/zh-cn/learn/css/building_blocks/the_box_model/index.md:198]:
```suggestion
[浏览器开发者工具](/zh-CN/docs/Learn/Common_questions/Tools_and_setup/What_are_browser_developer_tools)可以使你更容易地理解盒模型。如果你在 Firefox 的 DevTools 中查看一个元素，你可以看到元素的大小以及它的外边距、内边距和边框。这是一个很好的检查元素大小的方式，可以便捷的判断你的盒子大小是否符合预期！
```

File: files/zh-cn/learn/css/building_blocks/the_box_model/index.md [hunk 3]
```diff
@@ -253,62 +272,64 @@ html {
...
-使用 `.links-list a` 选择器将 `display: inline-block` 添加到样式规则中，你将看到它是如何通过内边距推开其他元素来修复这个问题的。
+## 技能测试！
 
-{{EmbedGHLiveSample("css-examples/learn/box-model/inline-block-nav.html", '100%', 600)}}
+本文已经结束，你还记得最重要的信息吗？在继续学习之前，可以找到一些进一步的测试来验证你是否保留了这些信息——请参阅[技能测试：盒模型](/zh-CN/docs/Learn/CSS/Building_blocks/Box_Model_Tasks)。
 
 ## 总结
 
-这就是你需要了解的关于盒子模型的大部分内容。如果以后你发现对于盒模型的布局仍有困惑，你将会回来温故这些内容。
+这就是你需要了解的关于盒子模型的大部分内容。今后，如果你对布局中盒子的大小感到困惑，不妨再来学习这一课。
```
Comment (awxiaoxian2020) [inline files/zh-cn/learn/css/building_blocks/the_box_model/index.md:331]:
```suggestion
这就是你需要了解的关于盒子模型的大部分内容。今后，如果你对布局中盒子的大小感到困惑，不妨再回来看看这些内容。
```

**Review by awxiaoxian2020 — CHANGES_REQUESTED** [review]

File: files/zh-cn/learn/css/building_blocks/the_box_model/index.md [hunk 4]
```diff
@@ -1,244 +1,263 @@
...
-最后设置两个段落为 `display: inline`。inline flex 容器和段落在一行上而不是像块级元素一样换行。
+在下一个示例中，我们可以看到 `inline` 元素是如何表现的。
 
-**你可以修改 `display: inline` 为 `display: block` 或者 `display: inline-flex` 改为 `display: flex` 来观察显示模式切换。**
+- 第一段中的 `<span>` 元素默认为行级，因此不会强制换行。
+- 设置为 `display: inline-flex` 的 `<ul>` 元素会创建一个行向盒子，其中包含一些弹性项目。
+- 这两个段落都设置为 `display: inline`。行向弹性容器和段落都在一行中流动，而不是分成两行（如果它们显示为块级元素，就会这样）。
 
-{{EmbedGHLiveSample("css-examples/learn/box-model/inline.html", '100%', 1000)}}
+**要在显示模式之间切换，可以将 `display: inline` 更改为 `display: block`，或将 `display: inline-flex` 更改为 `display: flex`** 。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/the_box_model/index.md:93]:
```suggestion
**要在显示模式之间切换，可以将 `display: inline` 更改为 `display: block`，或将 `display: inline-flex` 更改为 `display: flex`。**
```

File: files/zh-cn/learn/css/building_blocks/the_box_model/index.md [hunk 5]
```diff
@@ -1,244 +1,263 @@
...
+**你能改变第二个盒子的大小（通过添加 CSS 到 `.alternate` 类中) 让它和第一个盒子宽高一样吗？**
 
-{{EmbedGHLiveSample("css-examples/learn/box-model/box-models.html", '100%', 1000)}}
+{{EmbedGHLiveSample("css-examples/learn/box-model/box-models.html", '100%', 1100)}}
 
-> **备注：** [单击此处查看答案](https://github.com/mdn/css-examples/blob/master/learn/solutions.md#the-box-model)。
+> **备注：** 在[这里](https://github.com/mdn/css-examples/blob/master/learn/solutions.md#the-box-model)查看该任务的解答。
 
-### 使用调试工具来查看盒模型
+### 使用浏览器调试工具来查看盒模型
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/the_box_model/index.md:196]:
```suggestion
### 使用浏览器开发者工具来查看盒模型
```

File: files/zh-cn/learn/css/building_blocks/the_box_model/index.md [hunk 6]
```diff
@@ -1,244 +1,263 @@
...
 
-[浏览器开发者工具](/zh-CN/docs/Learn/Common_questions/What_are_browser_developer_tools)可以使你更容易地理解 box 模型。如果你在 Firefox 的 DevTools 中查看一个元素，你可以看到元素的大小以及它的外边距、内边距和边框。这是一个很好的检查元素大小的方式，可以便捷的判断你的盒子大小是否符合预期 !
+[浏览器开发者工具](/zh-CN/docs/Learn/Common_questions/Tools_and_setup/What_are_browser_developer_tools)可以使你更容易地理解盒模型。如果你在 Firefox 的 DevTools 中查看一个元素，你可以看到元素的大小以及它的外边距、内边距和边框。这是一个很好的检查元素大小的方式，可以便捷的判断你的盒子大小是否符合预期！
 
-![Inspecting the box model of an element using Firefox DevTools](box-model-devtools.png)
+![使用 Firefox DevTools 检查元素的盒模型](box-model-devtools.png)
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/the_box_model/index.md:200]:
```suggestion
![使用 Firefox 开发者工具检查元素的盒模型](box-model-devtools.png)
```

File: files/zh-cn/learn/css/building_blocks/the_box_model/index.md [hunk 7]
```diff
@@ -1,244 +1,263 @@
...
 
-## 外边距，内边距，边框
+## 外边距、内边距和边框
 
-你已经在上面的示例中看到了{{cssxref("margin")}}、{{cssxref("padding")}}和{{cssxref("border")}}属性。该示例中使用的是属性的**简写**，允许我们一次设置盒子的四个边。这些简写等价于分别控制盒子的不同边的普通写法。
+在上面的示例中，你已经看到了{{cssxref("margin")}}、{{cssxref("padding")}}和{{cssxref("border")}}属性的作用。该示例中使用了**简写属性**，允许我们一次性设置盒子的所有边。这些简写属性也有等效的普通属性，可以单独控制盒子的不同边。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/the_box_model/index.md:204]:
```suggestion
在上面的示例中，你已经看到了 {{cssxref("margin")}}、{{cssxref("padding")}} 和 {{cssxref("border")}} 属性的作用。该示例中使用了**简写属性**，允许我们一次性设置盒子的所有边。这些简写属性也有等效的普通属性，可以单独控制盒子的不同边。
```

File: files/zh-cn/learn/css/building_blocks/the_box_model/index.md [hunk 8]
```diff
@@ -1,244 +1,263 @@
...
+
+- 两个正外边距将合并为一个外边距。其大小等于最大的单个外边距。
+- 两个负外边距会折叠，并使用最小（离零最远）的值。
+- 如果其中一个外边距为负值，其值将从总值中*减*去。
 
-在下面的例子中，我们有两个段落。顶部段落的页 `margin-bottom`为 50px。第二段的`margin-top` 为 30px。因为外边距折叠的概念，所以框之间的实际外边距是 50px，而不是两个外边距的总和。
+在下面的示例中，我们有两个段落。最上面一段的 `margin-bottom` 为 50 像素，另一段的 `margin-top` 为 30 像素。页边距折叠在一起，因此方框之间的实际页边距是 50 像素，而不是两个页边距的总和。
 
-**你可以通过将第 2 段的 `margin-top` 设置为 0 来测试它。两个段落之间的可见边距不会改变——它保留了第一个段落 `margin-bottom`设置的 50 像素。**
+**你可以通过将第 2 段的 `margin-top` 设置为 0 来测试它。两个段落之间的可见边距不会改变——它保留了第一个段落 `margin-bottom` 设置的 50 像素。如果将其设置为 -10px，你会发现总边距变成了 40px（从 50px 中减去该负值）**
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/the_box_model/index.md:233]:
```suggestion
**你可以通过将第 2 段的 `margin-top` 设置为 0 来测试它。两个段落之间的可见边距不会改变——它保留了第一个段落 `margin-bottom` 设置的 50 像素。如果将其设置为 -10px，你会发现总边距变成了 40px（从 50px 中减去该负值）。**
```

File: files/zh-cn/learn/css/building_blocks/the_box_model/index.md [hunk 9]
```diff
@@ -1,244 +1,263 @@
...
+{{EmbedGHLiveSample("css-examples/learn/box-model/margin.html", '100%', 800)}}
 
 #### 外边距折叠
 
-理解外边距的一个关键是外边距折叠的概念。如果你有两个外边距相接的元素，这些外边距将合并为一个外边距，即最大的单个外边距的大小。
+根据外边距相接触的两个元素是正边距还是负边距，结果会有所不同：
+
+- 两个正外边距将合并为一个外边距。其大小等于最大的单个外边距。
+- 两个负外边距会折叠，并使用最小（离零最远）的值。
+- 如果其中一个外边距为负值，其值将从总值中*减*去。
```
Comment (yin1999) [inline files/zh-cn/learn/css/building_blocks/the_box_model/index.md:229]:
```suggestion
- 如果其中一个外边距为负值，其值将从总值中*减去*。
```

**Review by yin1999 — COMMENTED** [review]
Some suggestions:

**Review by yin1999 — APPROVED** [review]
Thank you @jasonren0403, @awxiaoxian2020

**Pull request merged**